cpp/src/arrow/builder.h

```diff
@@ -674,6 +673,10 @@ class ARROW_EXPORT FixedSizeBinaryBuilder : public ArrayBuilder {
                          MemoryPool* pool ARROW_MEMORY_POOL_DEFAULT);
 
   Status Append(const uint8_t* value);
+
+  template <size_t NBYTES>
+  Status Append(const std::array<uint8_t, NBYTES>& data);
```

Comment:
So I guess if you try using this for any value of NBYTES other than 16 you'll get a linker error?

Reply:
Good point. Let me move that implementation to the header.